kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/protocolfilter/KafkaServiceReconcilerTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.protocolfilter;
+
+import java.time.Clock;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import io.fabric8.kubernetes.client.server.mock.KubernetesMockServer;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaServiceBuilder;
+import io.kroxylicious.kubernetes.operator.kafkaservice.KafkaServiceReconciler;
+
+import static io.kroxylicious.kubernetes.operator.assertj.OperatorAssertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnableKubernetesMockClient(crud = true)
+@ExtendWith(MockitoExtension.class)
+class KafkaServiceReconcilerTest {
+
+    public static final long OBSERVED_GENERATION = 1345L;
+    KubernetesClient kubeClient;
+    KubernetesMockServer mockServer;
+
+    @Mock
+    Context<KafkaService> context;
+
+    private KafkaServiceReconciler kafkaProtocolFilterReconciler;
+
+    @BeforeEach
+    void setUp() {
+        kafkaProtocolFilterReconciler = new KafkaServiceReconciler(Clock.systemUTC());
+    }
+
+    @Test
+    void shouldMarkFilterAccepted() throws Exception {
+        // Given
+        final KafkaService kafkaService = new KafkaServiceBuilder().withNewStatus().withObservedGeneration(OBSERVED_GENERATION).endStatus().build();
+
+        // When
+        final UpdateControl<KafkaService> updateControl = kafkaProtocolFilterReconciler.reconcile(kafkaService, context);
+
+        // Then
+        assertThat(updateControl)
```

Comment:
Maybe we should write a common `UpdateControlAssert` ?

Reply:
Probably. We want to take another run at a common status class and assert as well